### PR TITLE
Add more fine grained pecentiles add 98, 99.5,99.6, 99.7, 99.8, 99.9

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe or @geirst PR

### CODE REVIEW

File: fbench/src/fbench/fbench.cpp [hunk 1]
```diff
@@ -250,24 +243,24 @@ FBench::PrintSummary()
...
-        printf("95 percentile:          %8.2f ms (approx)\n", p95);
-    else         printf("95 percentile:          %8.2f ms\n", p95);
-    if (p99 > status._timetable.size() / status._timetableResolution - 1)
-        printf("99 percentile:          %8.2f ms (approx)\n", p99);
-    else         printf("99 percentile:          %8.2f ms\n", p99);
+
+    for (double percentile : {25.0, 50.0, 75.0, 90.0, 98.0, 99.0, 99.5, 99.6, 99.7, 99.8, 99.9}) {
+        if (percentile <= 99.0) {
+            //Backwards compatible printing
+            if (percentile > (status._timetable.size() / status._timetableResolution - 1)) {
```
Comment (havardpe) [inline fbench/src/fbench/fbench.cpp:250]: you need to check the result of the percentile calculation against the sampled data to see if it is approximate, not the input parameter of the percentile calculation...

File: fbench/src/fbench/fbench.cpp [hunk 2]
```diff
@@ -250,24 +243,24 @@ FBench::PrintSummary()
...
+                printf("%2d percentile:          %8.2f ms (approx)\n", int(percentile), status.GetPercentile(percentile));
+            } else {
+                printf("%2d percentile:          %8.2f ms\n", int(percentile), status.GetPercentile(percentile));
+            }
+        } else {
+            if (percentile > (status._timetable.size() / status._timetableResolution - 1)) {
+                printf("%2.1f percentile:          %8.2f ms (approx)\n", percentile, status.GetPercentile(percentile));
```
Comment (havardpe) [inline fbench/src/fbench/fbench.cpp:257]: factoring out the formatting of the percentile number would probably be useful here...

File: fbench/src/fbench/fbench.cpp (hunk 1 above)
Comment (baldersheim) [reply fbench/src/fbench/fbench.cpp]: Good catch, fixed

File: fbench/src/fbench/fbench.cpp (hunk 2 above)
Comment (baldersheim) [reply fbench/src/fbench/fbench.cpp]: Factored out both optional (approx) and formating of percentile.

**Comment by baldersheim** [comment]
@havardpe Now you have another look.

File: fbench/src/fbench/fbench.cpp [hunk 3]
```diff
@@ -200,6 +200,28 @@ FBench::StopClients()
...
+    return (latency > (status._timetable.size() / status._timetableResolution - 1))
+           ? "ms (approx)"
+           : "ms";
+}
+
+std::string
+fmtPercentile(double percentile) {
+    char buf[32];
+    if (percentile <= 99.0) {
+        snprintf(buf, sizeof(buf), "%2d  ", int(percentile));
```
Comment (havardpe) [inline fbench/src/fbench/fbench.cpp:216]: I think this will add 2 extra spaces in the 'backwards' case...

**Review by havardpe — COMMENTED** [review]
looks good, but I think the old-style formatting is a bit off.

File: fbench/src/fbench/fbench.cpp (hunk 3 above)
Comment (baldersheim) [reply fbench/src/fbench/fbench.cpp:216]: Yes, that is fine. I just wanted to avid decimals  in the old percentiles

**Pull request merged**